aggregate/aggregate

```diff
@@ -0,0 +1,223 @@
+
+    # Load string2pdb output:
+    string_df = pd.read_csv(args.s)
+
+    
+    ## MENTHA2PDB OUTPUT PRE-PROCESSING: ##
+
+    if not mentha_df.empty:
+    
+        # Replace 'na' values with "":
+        mentha_df.replace('na', "", inplace=True)
+
+        # Keep only the required columns
+        mentha_df = mentha_df[[
+            "target uniprot id", 
+            "target uniprot gene", 
+            "interactor uniprot id", 
+            "interactor uniprot gene", 
+            "mentha score", 
+            "PDB id", 
+            "pDockQ HuMap", 
+            "pDockQ HuRI"
+        ]]
+
+        # Simplify gene names by keeping only the part before the first space or '{' :
+        # For cases like  MEOX2 {ECO:0000313|EMBL:CAG38790.1} or MEOX2{ECO:0000313|EMBL:CAG38790.1}
+        mentha_df["target uniprot gene"] = mentha_df["target uniprot gene"].str.replace(r"[ {].*", "", regex=True)
+        mentha_df["interactor uniprot gene"] = mentha_df["interactor uniprot gene"].str.replace(r"[ {].*", "", regex=True)
+
+        # Initialize the "structure" column with the PDB id values:
+        mentha_df["structure"] = mentha_df["PDB id"].fillna("")
+
+        # Add "AF_Huri_HuMAP" to "structure" if HuMap or HuRI values are not "":
+        mentha_df["structure"] = mentha_df.apply(
+            lambda row: ",".join(filter(None, [row["structure"], "AF_Huri_HuMAP"])) \
+            if row["pDockQ HuMap"] != "" or row["pDockQ HuRI"] != "" else row["structure"],
+            axis=1
+        )
+
+        # Drop columns "PDB id", "pDockQ HuMap", and "pDockQ HuRI":
+        mentha_df.drop(columns=["PDB id", "pDockQ HuMap", "pDockQ HuRI"], inplace=True)
+
+         # Group by all columns except "structure" and aggregate "structure" as comma-separated strings
+        mentha_df = mentha_df.groupby([
+            "target uniprot id", 
+            "target uniprot gene", 
+            "interactor uniprot id", 
+            "interactor uniprot gene", 
+            "mentha score"
+        ], as_index=False).agg({"structure": lambda x: ",".join(sorted(set(filter(None, x))))})
+
+        mentha_df.sort_values(by="mentha score", ascending=False, inplace=True)
+
+
+    ## STRING2PDB OUTPUT PRE-PROCESSING: ##
+
+    if not string_df.empty:
+
+        string_df.fillna("", inplace=True)
+
+        # Keep only required columns from string_df:
+        string_df = string_df[[
+            "Target_protein", 
+            "Target_Uniprot_AC", 
+            "Interactor", 
+            "Interactor_UniProt_AC", 
+            "String_score", 
+            "PDB_ID"
+        ]]
+
+        # Rename columns to match mentha for consistency:
+        string_df.rename(columns={
+            "Target_protein": "target uniprot gene",
+            "Target_Uniprot_AC": "target uniprot id",
+            "Interactor": "interactor uniprot gene",
+            "Interactor_UniProt_AC": "interactor uniprot id",
+            "String_score": "string score"
+        }, inplace=True)
```

Comment:
honestly I'd rather keep your column names since they are better and more accurate